source/common/upstream/upstream_impl.cc

```diff
@@ -214,7 +231,24 @@ ClusterImplBase::ClusterImplBase(const envoy::api::v2::Cluster& cluster,
                                  Runtime::Loader& runtime, Stats::Store& stats,
                                  Ssl::ContextManager& ssl_context_manager, bool added_via_api)
     : runtime_(runtime), info_(new ClusterInfoImpl(cluster, source_address, runtime, stats,
-                                                   ssl_context_manager, added_via_api)) {}
+                                                   ssl_context_manager, added_via_api)) {
+  priority_set_.addMemberUpdateCb([this](uint32_t priority,
+                                         const std::vector<HostSharedPtr>& hosts_added,
+                                         const std::vector<HostSharedPtr>& hosts_removed) {
+    // FIXME(alyssawilk) talk to Matt about how to change stats in prod.
```

Comment:
IMO it's probably better to just keep the current stats, and also have per-priority stats? (Perhaps only per-priority if there are > 1 priorities).

Reply:
SG.  I'll leave these as-is and we can add per-priority when/if folks want 'em

We may be able to get away with just having priority in LB reports, and I've added a TODO for that